File: spring-rest-configuration-autodiscover.scala/src/main/scala/demo/service/example/ExamplePropertyEnumService.scala
package demo.service.example

import demo.service.core.{Code, PropertyEnumService}
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class ExamplePropertyEnumService extends PropertyEnumService[ExampleEnum.ExampleEnum] {

  @Value("${example.property.enum}")
  private val defaultValue: String = null

  override def getCode: Code.Value = Code.EXAMPLE_ENUM_PROPERTY

  override def getDefaultValue: ExampleEnum.Value = ExampleEnum.withName(defaultValue)

  override def getEnumValues: Set[ExampleEnum.Value] = ExampleEnum.values
}

object ExampleEnum extends Enumeration {
  type ExampleEnum = Value
  val ONE, TWO, THREE = Value
}

File: spring-rest-configuration-autodiscover.java/src/main/java/demo/service/example/ExamplePropertyEnumService.java
package demo.service.example;

import demo.service.core.Code;
import demo.service.core.PropertyEnumService;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class ExamplePropertyEnumService extends PropertyEnumService<ExamplePropertyEnumService.ExampleEnum> {

    @Value("${example.property.enum}")
    private ExampleEnum defaultValue;

    @Override
    public Code getCode() {
        return Code.EXAMPLE_ENUM_PROPERTY;
    }

    @Override
    public ExampleEnum getDefaultValue() {
        return defaultValue;
    }

    @Override
    public Class<ExampleEnum> getEnumClass() {
        return ExampleEnum.class;
    }

    public enum ExampleEnum {
        ONE,
        TWO,
        THREE,
    }
}
